### DISCUSSION TITLE
Experimenting with ATtiny1626 and Buzzer on TCB

### DISCUSSION THREAD

**Discussion by GeorgeIoak** [SpenceKonde/megaTinyCore#598] [Q&A]
Before I reinvent the wheel I figured I'd ask here if anyone had experimented with different sounds on buzzers. I know you can set different frequencies to get different tones (within reason) but has anyone also experimented with changing the duty cycle?

I've done some initial tests with `tone` and `analogWrite` but it seems like I'm getting some inconsistent results and I'm not sure if that's a function of the buzzer or the way the code is working.

**Comment by apws** [comment]
well, over 30years ago on 8080 school computer I did some "synth" by cycling 8 or 16 bits on driven speaker and it definitely had different spectrum/tone, if you can find some FFT simulation on web (only idea) then it can be proven too - so lets try it, sure :)

**Comment by GeorgeIoak** [comment]
I don't think it has to be that fancy to use FFT, my main goal is to just sound a warning and as I started testing I began to wonder what different sounds I could create if I also varied the duty cycle on these buzzers. I like your thinking outside the box though!

**Reply by apws** [reply]
piezzo buzzer is probably too much "digital", though... and also "tuned" for some frequency band; having real tiny speaker with simple RC filter at least to limit harmonics up to say 10-15kHz will have some effect for sure - with FFT I mean to simulate/predict the effect, may be simplest in free AudaCity or Reaper editors - may be in some old mobile phone you can find speaker to test  ))

**Reply by GeorgeIoak** [reply]
Well it's definitely not anything pleasant to listen to but I just wanted to try something more than just a beep. Maybe someone will see this and chime in if they've messed around with these before otherwise I'll have to just dig under the hood a bit and run some tests.
